src/Integration.Vsix/Suppression/SuppressionManager.cs

```diff
@@ -0,0 +1,129 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using SonarLint.VisualStudio.Integration.Suppression;
+
+namespace SonarLint.VisualStudio.Integration.Vsix.Suppression
+{
+    internal sealed class SuppressionManager : IDisposable
+    {
+        private readonly IServiceProvider serviceProvider;
+        private readonly IActiveSolutionBoundTracker activeSolutionBoundTracker;
+
+        private DelegateInjector delegateInjector;
+        private LiveIssueFactory liveIssueFactory;
+        private ServerIssuesProvider serverIssueProvider;
+
+        public SuppressionManager(IServiceProvider serviceProvider)
+        {
+            this.serviceProvider = serviceProvider;
+            activeSolutionBoundTracker = serviceProvider.GetMefService<IActiveSolutionBoundTracker>();
+            activeSolutionBoundTracker.SolutionBindingChanged += OnSolutionBindingChanged;
+
+            RefreshSuppresionHandling();
+        }
+
+        private void RefreshSuppresionHandling()
+        {
+            if (activeSolutionBoundTracker.IsActiveSolutionBound)
+            {
+                SetupSuppressionHandling();
+            }
+            else
+            {
+                CleanupSuppressionHandling();
+            }
+        }
+
+        private void SetupSuppressionHandling()
+        {
+            liveIssueFactory = new LiveIssueFactory(serviceProvider);
```

Comment:
You don't have to change it but I tend to avoid creating new classes from within a class. I prefer using a factory (easier to mock and to provide a different behavior if needed).